driver-async/src/test/unit/com/mongodb/async/client/UnsynchronizedTestObserver.java

```diff
@@ -0,0 +1,263 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.async.client;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+
+public class UnsynchronizedTestObserver<T> implements Observer<T> {
+
+    private final Observer<T> delegate;
+    private final CountDownLatch latch = new CountDownLatch(1);
+    private final ArrayList<T> onNextEvents = new ArrayList<T>();
+    private final ArrayList<Throwable> onErrorEvents = new ArrayList<Throwable>();
+    private final ArrayList<Void> onCompleteEvents = new ArrayList<Void>();
+
+    private Subscription subscription;
+
+    public UnsynchronizedTestObserver(final Observer<T> delegate) {
+        this.delegate = delegate;
+    }
+
+    public UnsynchronizedTestObserver() {
+        this(new Observer<T>() {
+
+            @Override
+            public void onSubscribe(final Subscription subscription) {
+                // do nothing
+            }
+
+            @Override
+            public void onNext(final T result) {
+                // do nothing
+            }
+
+            @Override
+            public void onComplete() {
+                // do nothing
+            }
+
+            @Override
+            public void onError(final Throwable e) {
+                // do nothing
+            }
+        });
+    }
+
+    @Override
+    public synchronized void onSubscribe(final Subscription subscription) {
```

Comment:
It's also weird that there are synchronized methods on this class!

Reply:
File has been removed.